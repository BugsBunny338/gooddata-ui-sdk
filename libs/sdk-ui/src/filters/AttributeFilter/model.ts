// (C) 2007-2018 GoodData Corporation
import { IElementQueryResult } from "@gooddata/sdk-backend-spi";

export interface IAttributeDisplayForm {
    meta: {
        title: string;
        uri: string;
        identifier: string;
    };
}

export interface IAttributeElement {
    uri: string;
    title: string;
}

export interface IAttributeElementsChildren {
    validElements: IElementQueryResult;
    loadMore: () => void;
    isLoading: boolean;
    error: any;
}
