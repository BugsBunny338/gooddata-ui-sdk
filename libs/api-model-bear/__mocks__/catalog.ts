// (C) 2019 GoodData Corporation
import { GdcCatalog } from "../src";

export const catalogAttribute: GdcCatalog.ICatalogAttribute = {
    identifier: "attr",
    type: "attribute",
    title: "Attribute",
    summary: "",
    groups: [],
    production: true,
    links: {
        defaultDisplayForm: "/gdc/md/project/obj/1",
        self: "/gdc/md/project/obj/0",
    },
};

export const catalogMetric: GdcCatalog.ICatalogMetric = {
    identifier: "measure",
    type: "metric",
    title: "Measure",
    summary: "",
    groups: [],
    production: true,
    expression: "SELECT SUM([/gdc/md/project/obj/4])",
    format: "#,##",
    links: {
        self: "/gdc/md/project/obj/2",
    },
};

export const catalogFact: GdcCatalog.ICatalogFact = {
    identifier: "fact",
    type: "fact",
    title: "Fact",
    summary: "",
    groups: [],
    production: true,
    links: {
        self: "/gdc/md/project/obj/3",
    },
};
