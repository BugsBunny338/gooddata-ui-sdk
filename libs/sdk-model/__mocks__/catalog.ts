// (C) 2019 GoodData Corporation
import { ICatalogAttribute, ICatalogMeasure, ICatalogFact, ICatalogDateDataset } from "../src";

export const catalogAttribute: ICatalogAttribute = {
    id: "attr",
    uri: "/attributes/attr",
    type: "attribute",
    title: "Attribute",
    description: "",
    groups: [],
    production: true,
    defaultDisplayForm: {
        id: "label.attr",
        description: "",
        production: true,
        title: "Attribute Display Form",
        uri: "/labels/label",
    },
};

export const catalogMeasure: ICatalogMeasure = {
    id: "measure",
    uri: "/measures/measure",
    type: "measure",
    title: "Measure",
    description: "",
    groups: [],
    production: true,
    expression: "SELECT SUM([/attributes/attr])",
    format: "#,##",
};

export const catalogFact: ICatalogFact = {
    id: "fact",
    uri: "/facts/fact",
    type: "fact",
    title: "Fact",
    description: "",
    groups: [],
    production: true,
};

export const catalogDateDataset: ICatalogDateDataset = {
    id: "dateDataset",
    uri: "/dateDatasets/dateDataset",
    type: "dateDataset",
    title: "Date Dataset",
    description: "",
    production: true,
    dateAttributes: [],
    relevance: 0,
};
