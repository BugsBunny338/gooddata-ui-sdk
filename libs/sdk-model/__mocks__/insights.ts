// (C) 2019 GoodData Corporation
import { SortItem, VisualizationProperties } from "../src";
import { IBucket } from "../src/execution/buckets";
import { IInsight } from "../src/insight";
import { IFilter } from "../src/execution/filter";
import identity = require("lodash/identity");

/*
 * Factory & builder for insight instances. Keeping it in test infrastructure for now, will see later on
 * whether we should move it to prod code and expose on public API.
 */

export function newInsight(visClassId: string, modifications: InsightsModifications = identity): IInsight {
    const builder = new InsightBuilder(visClassId);

    return modifications(builder).build();
}

export type InsightsModifications = (builder: InsightBuilder) => InsightBuilder;

export class InsightBuilder {
    private insight: IInsight["insight"];

    constructor(visClassUri: string) {
        this.insight = {
            visualizationUrl: visClassUri,
            title: "Untitled",
            buckets: [],
            filters: [],
            sorts: [],
            properties: {},
            identifier: "random",
            uri: "random",
        };
    }

    public title = (title: string): InsightBuilder => {
        this.insight.title = title;

        return this;
    };

    public buckets = (buckets: IBucket[]): InsightBuilder => {
        this.insight.buckets = buckets;

        return this;
    };

    public filters = (filters: IFilter[]): InsightBuilder => {
        this.insight.filters = filters;

        return this;
    };

    public sorts = (sorts: SortItem[]): InsightBuilder => {
        this.insight.sorts = sorts;

        return this;
    };

    public properties = (properties: VisualizationProperties): InsightBuilder => {
        this.insight.properties = properties;

        return this;
    };

    public isLocked = (isLocked: boolean): InsightBuilder => {
        this.insight.isLocked = isLocked;

        return this;
    };

    public updated = (updated: string): InsightBuilder => {
        this.insight.updated = updated;

        return this;
    };

    public build = (): IInsight => {
        return {
            insight: this.insight,
        };
    };
}
