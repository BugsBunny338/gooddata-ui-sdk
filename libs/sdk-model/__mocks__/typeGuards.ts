// (C) 2019 GoodData Corporation

/**
 * Contains
 */
export const InvalidInputTestCases: Array<[boolean, string, any]> = [
    [false, "null", null],
    [false, "undefined", undefined],
    [false, "empty object", {}],
    [false, "array", []],
    [false, "string", "bleh"],
];
