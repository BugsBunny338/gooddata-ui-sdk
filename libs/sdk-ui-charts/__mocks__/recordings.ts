// (C) 2020 GoodData Corporation

import { DataViewFacade } from "@gooddata/sdk-backend-spi";
import { ScenarioRecording, recordedDataView } from "@gooddata/sdk-backend-mockingbird";

export function recordedDataFacade(rec: ScenarioRecording): DataViewFacade {
    return new DataViewFacade(recordedDataView(rec));
}
