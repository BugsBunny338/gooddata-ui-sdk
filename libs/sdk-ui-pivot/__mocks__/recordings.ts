// (C) 2020 GoodData Corporation

import { DataViewFacade } from "@gooddata/sdk-backend-spi";
import { ScenarioRecording, recordedDataView, DataViewAll } from "@gooddata/sdk-backend-mockingbird";

export function recordedDataFacade(rec: ScenarioRecording, dataViewId: string = DataViewAll): DataViewFacade {
    return new DataViewFacade(recordedDataView(rec, dataViewId));
}
