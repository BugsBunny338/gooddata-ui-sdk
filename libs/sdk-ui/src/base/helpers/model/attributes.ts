// (C) 2018 GoodData Corporation
import { getQualifierObject } from "./utils";
import { IAttribute } from "@gooddata/sdk-model";

export class AttributeBuilder implements IAttribute {
    private static lastAttributeId = 0;
    public attribute: IAttribute["attribute"];

    constructor(displayForm: string) {
        this.attribute = {
            displayForm: getQualifierObject(displayForm),
            localIdentifier: `va_${AttributeBuilder.lastAttributeId++}`,
        };
    }

    public alias = (alias: string) => {
        this.attribute.alias = alias;
        return this;
    };

    public localIdentifier = (localIdentifier: string) => {
        this.attribute.localIdentifier = localIdentifier;
        return this;
    };
}

export const attribute = (displayForm: string) => new AttributeBuilder(displayForm);
