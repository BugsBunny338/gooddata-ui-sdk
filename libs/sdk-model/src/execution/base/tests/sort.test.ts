// (C) 2019 GoodData Corporation

import {
    IAttributeSortItem,
    IMeasureSortItem,
    newAttributeLocator,
    newAttributeSort,
    newMeasureSort,
    sortEntityIds,
} from "../sort";
import { Account, Won } from "../../../../__mocks__/model";
import { InvariantError } from "ts-invariant";

const AttributeSort: IAttributeSortItem = newAttributeSort(Account.Default, "asc");
const MeasureSort1: IMeasureSortItem = newMeasureSort(Won, "asc");
const MeasureSort2: IMeasureSortItem = newMeasureSort(Won, "asc", [
    newAttributeLocator(Account.Default, "value"),
]);

describe("sortEntityIds", () => {
    const Scenarios: Array<[string, any]> = [
        ["get local ids from attribute sort", AttributeSort],
        ["get local ids from measure sort with just measure locator", MeasureSort1],
        ["get local ids from measure sort with measure locator and attribute locators", MeasureSort2],
    ];

    it.each(Scenarios)("should %s", (_desc, input) => {
        expect(sortEntityIds(input)).toMatchSnapshot();
    });

    const InvalidScenarios: Array<[string, any]> = [
        ["get empty result if input undefined", undefined],
        ["get empty result if input null", null],
    ];

    it.each(InvalidScenarios)("should %s", (_desc, input) => {
        expect(() => sortEntityIds(input)).toThrow();
    });
});

describe("newAttributeSort", () => {
    const Scenarios: Array<[string, any, any, any]> = [
        ["create asc sort by default", Account.Default, undefined, undefined],
        ["create asc sort for attribute object", Account.Default, "asc", undefined],
        ["create desc sort for attribute object", Account.Default, "desc", undefined],
        ["create asc sort for attribute", "localId1", "asc", undefined],
        ["create desc sort for attribute", "localId1", "desc", undefined],
        ["create sort with aggregation", Account.Default, "asc", true],
    ];

    it.each(Scenarios)("should %s", (_desc, attrArg, sortArg, aggArg) => {
        expect(newAttributeSort(attrArg, sortArg, aggArg)).toMatchSnapshot();
    });

    it("should throw if attribute input is undefined", () => {
        expect(() => newAttributeSort(undefined as any, "asc")).toThrowError(InvariantError);
    });

    it("should throw if attribute input is null", () => {
        expect(() => newAttributeSort(undefined as any, "asc")).toThrowError(InvariantError);
    });
});

describe("newMeasureSort", () => {
    const Scenarios: Array<[string, any, any, any]> = [
        ["create asc sort by default", Won, undefined, undefined],
        ["create asc sort for measure object", Won, "asc", undefined],
        ["create desc sort for measure object", Won, "desc", undefined],
        ["create asc sort for measure", "localId1", "asc", undefined],
        ["create desc sort for measure", "localId1", "desc", undefined],
        [
            "create desc sort for measure and attribute locators",
            "localId1",
            "desc",
            [newAttributeLocator(Account.Default, "value")],
        ],
    ];

    it.each(Scenarios)("should %s", (_desc, measureArg, sortArg, attrArg) => {
        expect(newMeasureSort(measureArg, sortArg, attrArg)).toMatchSnapshot();
    });

    it("should throw if measure input is undefined", () => {
        expect(() => newMeasureSort(undefined as any)).toThrowError(InvariantError);
    });

    it("should throw if measure input is undefined", () => {
        expect(() => newMeasureSort(undefined as any)).toThrowError(InvariantError);
    });
});

describe("newAttributeLocator", () => {
    const Scenarios: Array<[string, any, any]> = [
        ["create locator for attr object", Account.Default, "value"],
        ["create locator for attr", "localId", "value"],
    ];

    it.each(Scenarios)("should %s", (_desc, attrArg, elementArg) => {
        expect(newAttributeLocator(attrArg, elementArg)).toMatchSnapshot();
    });

    const InvalidScenarios: Array<[string, any, any]> = [
        ["attribute undefined", undefined, "value"],
        ["attribute null", null, "value"],
        ["element undefined", Account.Default, undefined],
        ["element null", Account.Default, null],
    ];

    it.each(InvalidScenarios)("should throw InvariantError when %s", (_desc, attrArg, elementArg) => {
        expect(() => newAttributeLocator(attrArg, elementArg)).toThrowError(InvariantError);
    });
});
