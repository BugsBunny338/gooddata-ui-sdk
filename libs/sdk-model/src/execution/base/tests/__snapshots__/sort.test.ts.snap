// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newAttributeAreaSort should create asc area sort by default 1`] = `
Object {
  "attributeSortItem": Object {
    "aggregation": "sum",
    "attributeIdentifier": "a_label.account.id",
    "direction": "asc",
  },
}
`;

exports[`newAttributeAreaSort should create asc area sort for attribute 1`] = `
Object {
  "attributeSortItem": Object {
    "aggregation": "sum",
    "attributeIdentifier": "localId1",
    "direction": "asc",
  },
}
`;

exports[`newAttributeAreaSort should create asc area sort for attribute object 1`] = `
Object {
  "attributeSortItem": Object {
    "aggregation": "sum",
    "attributeIdentifier": "a_label.account.id",
    "direction": "asc",
  },
}
`;

exports[`newAttributeAreaSort should create desc area sort for attribute 1`] = `
Object {
  "attributeSortItem": Object {
    "aggregation": "sum",
    "attributeIdentifier": "localId1",
    "direction": "desc",
  },
}
`;

exports[`newAttributeAreaSort should create desc area sort for attribute object 1`] = `
Object {
  "attributeSortItem": Object {
    "aggregation": "sum",
    "attributeIdentifier": "a_label.account.id",
    "direction": "desc",
  },
}
`;

exports[`newAttributeLocator should create locator for attr 1`] = `
Object {
  "attributeLocatorItem": Object {
    "attributeIdentifier": "localId",
    "element": "value",
  },
}
`;

exports[`newAttributeLocator should create locator for attr object 1`] = `
Object {
  "attributeLocatorItem": Object {
    "attributeIdentifier": "a_label.account.id",
    "element": "value",
  },
}
`;

exports[`newAttributeSort should create asc sort by default 1`] = `
Object {
  "attributeSortItem": Object {
    "attributeIdentifier": "a_label.account.id",
    "direction": "asc",
  },
}
`;

exports[`newAttributeSort should create asc sort for attribute 1`] = `
Object {
  "attributeSortItem": Object {
    "attributeIdentifier": "localId1",
    "direction": "asc",
  },
}
`;

exports[`newAttributeSort should create asc sort for attribute object 1`] = `
Object {
  "attributeSortItem": Object {
    "attributeIdentifier": "a_label.account.id",
    "direction": "asc",
  },
}
`;

exports[`newAttributeSort should create desc sort for attribute 1`] = `
Object {
  "attributeSortItem": Object {
    "attributeIdentifier": "localId1",
    "direction": "desc",
  },
}
`;

exports[`newAttributeSort should create desc sort for attribute object 1`] = `
Object {
  "attributeSortItem": Object {
    "attributeIdentifier": "a_label.account.id",
    "direction": "desc",
  },
}
`;

exports[`newMeasureSort should create asc sort by default 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "asc",
    "locators": Array [
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "m_afSEwRwdbMeQ",
        },
      },
    ],
  },
}
`;

exports[`newMeasureSort should create asc sort for measure 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "asc",
    "locators": Array [
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "localId1",
        },
      },
    ],
  },
}
`;

exports[`newMeasureSort should create asc sort for measure object 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "asc",
    "locators": Array [
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "m_afSEwRwdbMeQ",
        },
      },
    ],
  },
}
`;

exports[`newMeasureSort should create desc sort for measure 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "desc",
    "locators": Array [
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "localId1",
        },
      },
    ],
  },
}
`;

exports[`newMeasureSort should create desc sort for measure and attribute locators 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "desc",
    "locators": Array [
      Object {
        "attributeLocatorItem": Object {
          "attributeIdentifier": "a_label.account.id",
          "element": "value",
        },
      },
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "localId1",
        },
      },
    ],
  },
}
`;

exports[`newMeasureSort should create desc sort for measure object 1`] = `
Object {
  "measureSortItem": Object {
    "direction": "desc",
    "locators": Array [
      Object {
        "measureLocatorItem": Object {
          "measureIdentifier": "m_afSEwRwdbMeQ",
        },
      },
    ],
  },
}
`;

exports[`sortEntityIds should get local ids from attribute sort 1`] = `
Object {
  "allIdentifiers": Array [
    "a_label.account.id",
  ],
  "attributeIdentifiers": Array [
    "a_label.account.id",
  ],
  "measureIdentifiers": Array [],
}
`;

exports[`sortEntityIds should get local ids from measure sort with just measure locator 1`] = `
Object {
  "allIdentifiers": Array [
    "m_afSEwRwdbMeQ",
  ],
  "attributeIdentifiers": Array [],
  "measureIdentifiers": Array [
    "m_afSEwRwdbMeQ",
  ],
}
`;

exports[`sortEntityIds should get local ids from measure sort with measure locator and attribute locators 1`] = `
Object {
  "allIdentifiers": Array [
    "a_label.account.id",
    "m_afSEwRwdbMeQ",
  ],
  "attributeIdentifiers": Array [
    "a_label.account.id",
  ],
  "measureIdentifiers": Array [
    "m_afSEwRwdbMeQ",
  ],
}
`;
